src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpTagWriter.cs

```diff
@@ -95,18 +95,19 @@ internal struct OtlpTagWriterArrayState
         public int WritePosition;
     }
 
-    private sealed class OtlpArrayTagWriter : ArrayTagWriter<OtlpTagWriterArrayState>
+    internal sealed class OtlpArrayTagWriter : ArrayTagWriter<OtlpTagWriterArrayState>
     {
         [ThreadStatic]
-        private static byte[]? threadBuffer;
+        internal static byte[]? ThreadBuffer;
+        private const int MaxBufferSize = 2 * 1024 * 1024;
```

Comment:
@CodeBlanch Should we change this to readonly? With the current implementation, even if customers want to use a reflection hack to increase the size, const makes it impossible.

Reply:
🤷

We do have this: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExperimentalOptions.cs

You could define some experimental options to control the two buffer sizes. Users will be able to bind them to IConfiguration/envvars if needed. Better mechanism than reflection if you want to make it configurable.

Reply:
We could consider post RC release.